### ISSUE TITLE
urlArgs for plugins similar to require.js

### ISSUE THREAD

**Issue by iamdustan** [uxebu/bonsai#101]
Having `urlArgs` similar to requirejs for plugins would be awesome for development.

Instead I have concocted this awesome code:

``` javascript
plugins: (function () {
          var plugins = ['vendor/lodash.js',
                          'js/bonsai/layer.js',
                          'js/bonsai/product.js',
                          'js/bonsai/shapes/heart.js',
                          'js/bonsai/shapes/triangle.js',
                          'js/bonsai/shapes/skookum.js'];
          var date = '?'+ (+new Date);
          plugins.forEach(function (path, i) {
            plugins[i] = path + date;
          });
          return plugins;
       })()
```

(sure, I could’ve just done a `.map()` but that would be so overrated)

**Comment by davidaurelio** [comment]
Good point. We intend to unify `urls` and `plugins` and will add support for urlArgs together with that.

**Comment by iamdustan** [comment]
I was looking at the docs a bit more last night (and now that I am using plugins) was curious what the difference between `urls` and `plugins` was.

**Comment by davidaurelio** [comment]
The differences, at the moment, are the following special globals that can be used by plugins:
- assigning a property to `exports` will make a variable of that name accessible to bonsai movies
- `wait()` can be used to prevent further plugins and urls to be fetched – useful if a plugin needs to initialize itself asynchronously.
- `done()` can be used to continue plugin loading.

Thing is, we have been using `done` exclusively as callback for requirejs.

The vision is to provide a dev build of bonsai that includes requirejs, so that people can load stuff async at dev time. We will provide an example how to make it work.

`exports` will be replaced by something like `stage.env.expose({a: 1, b: 2, /* … */})`
